libs/wingsdk/src/target-sim/redis.inflight.ts

```diff
@@ -0,0 +1,79 @@
+import Dockerode from "dockerode";
+import { ISimulatorResourceInstance } from "./resource";
+import { RedisSchema } from "./schema-resources";
+import { IRedisClient } from "../redis";
+import { ISimulatorContext } from "../testing/simulator";
+
+// Issue using types from ioredis with JSII
+//import { Redis as IoRedis } from "ioredis";
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+const IoRedis = require("ioredis");
```

Comment:
Yeah, this is a tricky issue to deal with - I haven't been able to come up with too many great solutions for this. 😞  I just encountered it again, trying to add the `express` dependency to the simulator for use in the `Api` resource.

If you're feeling ambitious, one other workaround you can try is to install [@types/ioredis](https://www.npmjs.com/package/@types/ioredis?activeTab=versions) with an older version of the type definitions (tagged with TS 3.7 or something) into wingsdk and see if JSII will use that instead of the newer types that it can't understand. Or even to use an older version of ioredis, like the 4.x version.